pytorch_lightning/utilities/fetching.py

```diff
@@ -0,0 +1,140 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from collections.abc import Iterable, Iterator
+from typing import Any, Generator, List, Optional, Tuple
+
+from torch.utils.data.dataloader import DataLoader
+
+from pytorch_lightning.trainer.supporters import CombinedLoader
+from pytorch_lightning.utilities.apply_func import apply_to_collection
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+
+class AbstractFetcher(ABC):
```

Comment:
```suggestion
class AbstractDataFetcher(ABC):
```

Can we call it data-fetcher everywhere to make sure we always know what it fetches?

Reply:
good point !